### PR TITLE
Introduce pytest and add test for /v1/health

### PR DESCRIPTION
**Pull request by andreagrandi**
This PR introduces `pytest` and add an initial test for the `/v1/health` endpoint.
I still need to figure out how to generate a temporary db or test/mocked data so we can test the /v1/daily-reports` endpoint too.

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pytest==5.3.5
+requests==2.23.0
```
Comment (andreagrandi) [inline requirements-test.txt:2]: TIL: 
```
In [2]: from fastapi.testclient import TestClient
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-37f52789b9f0> in <module>
----> 1 from fastapi.testclient import TestClient

~/.virtualenvs/covid-api/lib/python3.8/site-packages/fastapi/testclient.py in <module>
----> 1 from starlette.testclient import TestClient  # noqa

~/.virtualenvs/covid-api/lib/python3.8/site-packages/starlette/testclient.py in <module>
     10 from urllib.parse import unquote, urljoin, urlsplit
     11
---> 12 import requests
     13
     14 from starlette.types import Message, Receive, Scope, Send

ModuleNotFoundError: No module named 'requests' to be able to import FastAPI test client.
```

We need to manually install `requests`:
```
(covid-api) ➜  covidapi git:(introducing-pytest) ✗ pip install requests
In [1]: from fastapi.testclient import TestClient

In [2]: quit()
```

**Review by MatMoore — APPROVED** [review]
There's a typo in the filename (test_healt.py) but otherwise LGTM 👍

**Comment by andreagrandi** [comment]
@MatMoore about test data / db, I just found this https://fastapi.tiangolo.com/advanced/testing-dependencies/
We are currently passing the `db` as dependency `def get_daily_reports(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):` so I suppose there could be a way to implement our own "fake" db which returns fixtures data and then inject the dependency during the tests like:
`app.dependency_overrides[common_parameters] = override_dependency`
But I'm not sure how to do that with an instance of a db. Can you help me to find out this, please? Once we know how to do it, it will be quite easy to replicate for every other endpoints.

**Comment by andreagrandi** [comment]
I got an idea:

```
def get_daily_reports(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    daily_reports = crud.get_daily_reports(db, skip=skip, limit=limit)
```

maybe we should not "mock" or inject the whole `db` object, but just the `crud` module? If we had something like this (we are in the `views.py` just to be clear):

```
from ..services.crud import CRUD  # we use a class here

def get_crud() -> CRUD:
    crud = CRUD()
    yield crud

def get_daily_reports(skip: int = 0, limit: int = 100, db: Session = Depends(get_db), crud_operations: CRUD = Depends(get_crud)):
    daily_reports = crud_operations.get_daily_reports(db, skip=skip, limit=limit)
```

we could inject the `CRUD` class and simply re-define the `get_daily_reports` method which would return a simple JSON dictionary of fake data.

**Comment by andreagrandi** [comment]
p.s: I reckon that in this way we are only testing the **views layer** and not the db / crud layer at all.

**Comment by andreagrandi** [comment]
This seems to be a solution to mock the whole db but I haven't tried it yet https://github.com/tiangolo/fastapi/issues/831
it's really weird that we have to define this custom stuff and it's not included somewhere in FastAPI 😓

**Comment by MatMoore** [comment]
I agree, fastapi is not making this easy...

I tried to implement what they suggested in the documentation (or at least, my interpretation of it) on this branch: https://github.com/andreagrandi/covid-api/compare/test-dependency-experiment#diff-28d3abd2cea435758f835a5f75cbc763R1

I didn't notice your last comment with that other example in, but I've just had a look at it and it's similar to what I've tried. Instead of creating a clean database I've gone with the approach of rolling back the transaction at the end of each test, which will be faster if there are lots of tests.

I think something along these lines will work (although it's more complicated that I'd hoped). We need to have some way of swapping out the database URL though otherwise the tests won't be repeatable.

I agree with your thinking about the other approach of injecting a CRUD object instead of the db session directly. I think we should test the db/crud layers because it's easy to make mistakes with SQLAlchemy 😕

**Comment by andreagrandi** [comment]
I'm going to merge this for now. 

Testing is important, but I would like to prioritise some work on data import and releasing at least a couple of endpoints asap.

We still don't know if and how other projects are going to use this. I would try to collect some real usage feedback as soon as possible, without making assumptions on how people will use the api or without making it super stable with still zero users.

I hope it makes sense.

**Comment by duganchen** [comment]
This is how I set up a test database:

https://github.com/tiangolo/fastapi/issues/831#issuecomment-601557441

Honestly, I think the FastAPI documentation assumes that you will not set up a test database, but rather just monkeypatch individual methods off crud.py so that your tests don't hit the database at all.

**Comment by MatMoore** [comment]
Thanks for sharing @duganchen! It's really helpful to have another example to work from 🙂

**Pull request merged**